WordPress/Sniffs/PHP/IniSetSniff.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPressCS\WordPress\Sniffs\PHP;
+
+use WordPressCS\WordPress\AbstractFunctionParameterSniff;
+use \PHP_CodeSniffer\Util\Tokens;
+
+/**
+ * Checks use of ini_set function with a blacklist for errors.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   x.x.x
+ */
+class IniSetSniff extends AbstractFunctionParameterSniff {
+
+	/**
+	 * Array of functions that must be checked
+	 *
+	 * @since x.x.x
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *     $target_functions = array(
+	 *         (string) Function name.
+	 *     );
+	 */
+	protected $target_functions = array(
+		'ini_set' => array(),
```

Comment:
The parameter value sniff was in part inspired by logic in `DeprecatedParametersSniff.php`. As it checks for the array key, I had to set it to something. To me it made more sense to make this an empty array, consistent with the other parameter checking sniffs, then to make the value `true`, or `null`. Which might imply it actually has a function or value. Also this seems more flexible if future edits are required, but this might be classic YAGNI.

Reply:
The `AbstractFunctionParameterSniff` which is underneath only requires the array keys to be the function names and a value to be set. It doesn't define the type of the value on purpose to give child classes maximum flexibility.
See: https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/develop/WordPress/AbstractFunctionParameterSniff.php#L35-L38

> To me it made more sense to make this an empty array, consistent with the other parameter checking sniffs

Generally speaking, for WPCS, if an array property is used where only the keys are relevant, the values are set to `true`.

This type of array format allows for using the faster `isset()` instead of `in_array()`, which would be needed if the array just contained values, or `array_key_exist()` if the value would be set to `null`.

See, for instance, the large number of array properties set up like this in the [`Sniff` class](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/develop/WordPress/Sniff.php).

For the sniffs based on the `AbstractFunctionParameterSniff` class, the values vary wildly depending on the use case of the child class:
* `PrefixAllGlobals` - uses the `Sniff::$hookInvokeFunctions` property where the values are set to `true` without meaning.
* `ValidHookName` - same as `PrefixAllGlobals`
* `PregQuoteDelimiter` - value set to `true` without meaning.
* `StrictInArray` - value set to boolean _with_ meaning. The meaning of the boolean is documented in the property documentation within the child sniff.
* `PluginMenuSlug` - value set to an array containing integers indicating the parameter positions which should be examined. Meaning documented in the property documentation.
* `I18nTextDomainFixer` - value set to the integer property position of the parameter to be examined.
* `DeprecatedParameters` - value set to a multi-dimensional array of parameter positions combined with info on that particular parameter.
* `DeprecatedParameterValues` - value set to a multi-dimensional array of parameter positions combined with info on that particular parameter.
* `DiscouragedConstants` - value set to the integer property position of the parameter to be examined.
* `EnqueuedResourceParameters` - value set to `true` without meaning.

As the value in the case of this sniff has no meaning, the most consistent choice based on the other code in WPCS, would be to use `true` as the value.

Reply:
Very valid, thanks for the great explanation. I've changed this part to be consistent with the other ones now.